laravel-app/public/assets/js/followUnfollow.js

```diff
@@ -0,0 +1,51 @@
+$(document).ready(function () {
+    $('.follow-button').each(function () {
+        const button = $(this);
+        const isFollowing = button.data('isFollowing') === true;
+        updateButtonText(button, isFollowing);
+    });
+
+    $(document).on('click', '.follow-button', function () {
+        const button = $(this);
+        const userId = button.data('user-id');
+        const isFollowing = button.data('isFollowing') === true;
+        const csrfToken = $('meta[name="csrf-token"]').attr('content');
+        const url = isFollowing ? button.data('delete-url') : button.data('url');
+
+        $.ajax({
+            url: url,
+            type: isFollowing ? 'DELETE' : 'PUT',
+            data: {
+                user: userId,
+                _token: csrfToken,
+            },
+            beforeSend: function () {
+                button.prop('disabled', true);
+            },
+            success: function () {
+                handleFollowUnfollow(button, isFollowing);
+                if (window.location.href.includes('/profile/')) {
+                    location.reload();
+                }
+            }
+        });
+    });
+});
+
+function handleFollowUnfollow(button, isFollowing) {
+    button.data('is-following', !isFollowing);
+
+    setTimeout(function () {
+        button.prop('disabled', false);
+        updateButtonText(button, !isFollowing);
+    }, 400);
```

Comment:
Why is did u add `setTimeout()` for this?

Reply:
I added the setTimeout to avoid button spams sir, the timer served as delay per click during which the button is disabled from clicks.

Reply:
Can you add a loader also for the button to create a better UX?

Reply:
I do have custom css for disabled buttons sir during setTimeout to make it visible that the buttons are unclickable

![settimeout to button](https://github.com/bpocrafael/microblog-project/assets/144191038/cf3f73f5-7890-47c3-b3df-c21dcc09e15c)